integration-tests/tests/test_cases.py

```diff
@@ -114,6 +119,43 @@ def setup_method(self):
         except (KeyError, json.JSONDecodeError):
             raise Exception(response.content)
 
+class EthApiTestCase(TransactionTestCase):
+    def get_block_header(self, block_number):
+        method = "ethclient_headerByNumber"
+        params = [self.network_id, block_number]
+        return self.rpc_valid_request(method, params, 13)
+
+    def get_transaction_receipt(self, tx_hash):
+        method = "ethclient_transactionReceipt"
+        params = [self.network_id, tx_hash]
+        return self.rpc_valid_request(method, params, 13)
+
+    def wait_until_tx_not_pending(self, tx_hash, timeout=10):
```

Comment:
Maybe there's some signal that can be used instead?

Reply:
hmm that would require initializing a bunch more stuff to get the pendingTxTracker working, I believe. Also we'll probably want faster rechecks for Anvil than in the real chains (every 10s). So, a considerable effort. Would you mind if I left that for a separate PR? Before answering, remember you just did this to me.
<img width="435" alt="image" src="https://github.com/user-attachments/assets/0fc531fa-c675-4d79-851d-af840127579e">

Reply:
Sure, we can do that 👍

Reply:
> we'll probably want faster rechecks for Anvil 

I think this was done, but I might confuse things

Reply:
> > we'll probably want faster rechecks for Anvil
> 
> I think this was done, but I might confuse things

I think we've got faster balance refreshes for Anvil, but I'm pretty sure we don't have faster Tx state checks.